### PR TITLE
[Agency Dashboard] Responsive layout for data viz page

### PR DESCRIPTION
**Pull request by terryttsai**
## Description of the change

Screen Shot 2022-11-17 at 4 24 44 PM
Screen Shot 2022-11-17 at 4 25 06 PM
Screen Shot 2022-11-17 at 4 25 18 PM

Responsive layout for desktop, tablet, and mobile screen sizes. 

Fixed ALL the resizing charts issues. I made it that on desktop, the chart resizes to match the height of the browser, but on tablet and mobile, since we have a single scrolling column I made the chart a fixed height.

I added in new Filter and Select Metric buttons for mobile, though the modals are not yet implemented!

## Type of change

> All pull requests must have at least one of the following labels applied (otherwise the PR will fail):

| Label                       	| Description                                                                                               	|
|-----------------------------	|-----------------------------------------------------------------------------------------------------------	|
| Type: Bug                   	| non-breaking change that fixes an issue                                                                   	|
| Type: Feature               	| non-breaking change that adds functionality                                                               	|
| Type: Breaking Change       	| fix or feature that would cause existing functionality to not work as expected                            	|
| Type: Non-breaking refactor 	| change addresses some tech debt item or prepares for a later change, but does not change functionality    	|
| Type: Configuration Change  	| adjusts configuration to achieve some end related to functionality, development, performance, or security 	|
| Type: Dependency Upgrade      | upgrades a project dependency - these changes are not included in release notes                             |

## Related issues

Closes #XXXX

### Development

**This box MUST be checked by the submitter prior to merging**:
- [ ] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [ ] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [ ] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: common/components/DataViz/DatapointsView.tsx [hunk 1]
```diff
@@ -183,10 +244,42 @@ export const DatapointsView: React.FC<{
...
+        <MobileSelectMetricsButton
+          onClick={() => setMobileSelectMetricsVisible(true)}
+        >
+          <GridIcon />
+          <SelectMetricsButtonText />
+        </MobileSelectMetricsButton>
+      </MobileSelectMetricsButtonContainer>
+      {mobileSelectMetricsVisible && (
+        <MobileSelectMetricsModalContainer>
+          {/* <MobileSelectMetricsModalTopRow>
```
Comment (lilidworkin) [inline common/components/DataViz/DatapointsView.tsx:270]: somewhat of a nit, but I generally advocate against checking in commented out code, and instead put all relevant details in a task and add a TODO (#task number) in the code instead

**Review by lilidworkin — APPROVED** [review]
I'm definitely out of my depth on this one -- @mxosman  should probably take a look too -- but generally I trust you @terryttsai! You can either merge this in as-is and address playtesting followups in the next PR, or you can make the changes to this PR, totally up to you.

File: agency-dashboard/src/DashboardView.tsx [hunk 2]
```diff
@@ -16,40 +16,48 @@
...
 } from "./DashboardView.styles";
 import { HeaderBar } from "./Header/HeaderBar";
 import { useStore } from "./stores";
 
-const LeftPanelBackButton = () => (
-  <LeftPanelBackButtonContainer>
+const getWidth = () =>
+  window.innerWidth ||
+  document.documentElement.clientWidth ||
+  document.body.clientWidth;
```
Comment (mxosman) [inline agency-dashboard/src/DashboardView.tsx:49]: tiny nit: maybe naming this something like `getScreenWidth` or something similar?

File: agency-dashboard/src/DashboardView.tsx [hunk 3]
```diff
@@ -151,38 +153,47 @@ const DashboardView = () => {
     <Container key={metricKey}>
       <HeaderBar />
       <LeftPanel>
-        <LeftPanelBackButton />
+        <BackButton onClick={() => navigate(`/agency/${agencyId}`)} />
```
Comment (mxosman) [inline agency-dashboard/src/DashboardView.tsx:156]: Wait - I don't think I realized you could pass the onClick function as a callback to a child component - I thought that it just gets triggered here. But, now that I'm looking at this - not sure what the difference would be having the onClick triggered here vs the container of the child component (`BackButtonContainer`)?

**Comment by mxosman** [comment]
Awesome seeing this come to life, Terry!

File: common/components/DataViz/DatapointsView.tsx (hunk 1 above)
Comment (terryttsai) [reply common/components/DataViz/DatapointsView.tsx:270]: Let me take this out for now!

File: agency-dashboard/src/DashboardView.tsx (hunk 3 above)
Comment (mxosman) [reply agency-dashboard/src/DashboardView.tsx:156]: That makes sense - thanks for clarifying!

**Pull request merged**